calyx-opt/src/passes/schedule_compaction.rs

```diff
@@ -58,53 +58,103 @@ impl Visitor for ScheduleCompaction {
 
         if let Ok(order) = algo::toposort(&total_order, None) {
             let mut total_time: u64 = 0;
-            let mut stmts: Vec<ir::StaticControl> = Vec::new();
 
+            // First we build the schedule.
             for i in order {
                 let mut start: u64 = 0;
                 for node in dependency.get(&i).unwrap() {
-                    let allow_start = schedule[node] + latency_map[node];
-                    if allow_start > start {
-                        start = allow_start;
-                    }
+                    start =
+                        std::cmp::max(start, schedule[node] + latency_map[node])
```

Comment:
This can be an iterator: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.max